commands/upload_integration_test.go

```diff
@@ -0,0 +1,68 @@
+package commands
+
+import (
+	"github.com/jfrog/jfrog-cli-core/utils/config"
+	"github.com/jfrog/jfrog-cli-core/utils/ioutils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func Test_UploadIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+	tests := []IntegrationTest{
+		{
+			Name: "Upload to specified target using target credentials",
+			Function: func(t *testing.T, rtDetails *config.ArtifactoryDetails,
+				targetRtDetails *config.ArtifactoryDetails) {
+				testBundle := getSupportBundle(t)
+				err := uploadSupportBundle(&HTTPClient{rtDetails: rtDetails, targetDetails: targetRtDetails},
+					&supportBundleCommandConfiguration{caseNumber: "foo"}, testBundle, "logs",
+					func() string { return "admin" })
+				assert.NoError(t, err)
+			},
+		},
+		{
+			Name: "Upload to default target without credentials",
+			Function: func(t *testing.T, rtDetails *config.ArtifactoryDetails,
+				targetRtDetails *config.ArtifactoryDetails) {
+				testBundle := getSupportBundle(t)
+				targetDetailsWithoutCreds := &config.ArtifactoryDetails{Url: targetRtDetails.Url}
+				err := uploadSupportBundle(&HTTPClient{rtDetails: rtDetails, targetDetails: targetDetailsWithoutCreds},
+					&supportBundleCommandConfiguration{
+						caseNumber:          "foo",
+						jfrogSupportLogsURL: targetRtDetails.GetUrl(),
+					}, testBundle, "logs", func() string { return "anonymous" })
+				assert.NoError(t, err)
+			},
+		},
+		{
+			Name: "Upload to non existent repository",
```

Comment:
Upload when target is offline

Reply:
Is it worth a test of upload to a non-existent repo?

Reply:
Good catch for the test name, I also added a test for non-existent repo. I'm not sure it adds much, but it doesn't cost much either.

Reply:
Actually different versions respond differently to uploading to a non-existent repo. Some return 403, others 405. Let's give this one a miss.